### ISSUE TITLE
Unable to retool 2 binaries from the same product

### ISSUE THREAD

**Issue by gaffo** [twitchtv/retool#37]
Twirp generators are a good example

**Comment by spenczar** [comment]
Right, this is because we *immediately* checkout out the version we want of a particular binary, right after `go get`-ing it.

We might need to first `go get` each binary, then go into each repo and `git checkout`, and then finally do a last pass at `go get` to fetch dependencies once we're on the right revision.

This is probably harder than it sounds: we have a list of packages; we don't necessarily know just from looking at the packages whether two of them are in the same repo. We'd have to figure that out.
